nix/shell.go

```diff
@@ -154,7 +159,22 @@ func writeDevboxShellrc(userShellrcPath string, userHook string) (path string, e
 		userShellrc = []byte{}
 	}
 
-	path = filepath.Join(tmp, filepath.Base(userShellrcPath))
+	var envPath []string
+	for _, kv := range env {
+		key, val, _ := strings.Cut(kv, "=")
+		if key == "PATH" {
+			envPath = filepath.SplitList(val)
+			break
+		}
+	}
```

Comment:
This part actually goes away in #124, which is based off of this PR. It was just a bit cumbersome to backport it here.